File: src/ServerPulse.Frontend/src/app/modules/analyzer/domain/dtos/responses/events/loadEventResponse.ts
import { BaseEventResponse, LoadEvent, mapBaseEventResponseToBaseEvent } from "../../../..";
import { TimeSpan } from "../../../../../shared";

export interface LoadEventResponse extends BaseEventResponse {
    endpoint: string;
    method: string;
    statusCode: number;
    duration: string;
    timestampUTC: Date;
}

export function mapLoadEventResponseToLoadEvent(response: LoadEventResponse): LoadEvent {
    return {
        ...mapBaseEventResponseToBaseEvent(response),
        endpoint: response?.endpoint,
        method: response?.method,
        statusCode: response?.statusCode,
        duration: TimeSpan.fromString(response?.duration),
        timestampUTC: new Date(response?.timestampUTC)
    };
}

File: src/ServerPulse.Backend/AnalyzerApi.Core/Dtos/Responses/Events/LoadEventResponse.cs
namespace AnalyzerApi.Core.Dtos.Responses.Events
{
    public class LoadEventResponse : BaseEventResponse
    {
        public string? Endpoint { get; set; }
        public string? Method { get; set; }
        public int StatusCode { get; set; }
        public TimeSpan Duration { get; set; }
        public DateTime TimestampUTC { get; set; }
    }
}